parser-typechecker/src/Unison/Lexer.hs

```diff
@@ -882,6 +908,18 @@ lexemes' eof = P.optional space >> do
 
     close = close' Nothing
 
+    blockDelimiter :: [String] -> P String -> P [Token Lexeme]
+    blockDelimiter open closeP = do
+      (pos1, close, pos2) <- positioned $ closeP
+      env                 <- S.get
+      case findClose open (layout env) of
+        Nothing -> err pos1 (UnexpectedDelimiter (quote close))
+          where quote s = "'" <> s <> "'"
+        Just (block, n) -> do
+          S.put (env { layout = drop n (layout env), opening = Just block })
```

Comment:
Did you try just -  

```suggestion
          S.put (env { layout = drop (n-1) (layout env) })
```

The existing line is actually popping that block from the layout stack, then later re-adding it back (with a newly updated starting column), which seems goofy and is probably subtly wrong for error reporting.